src/java/org/apache/cassandra/index/sai/disk/v1/V1OnDiskFormat.java

```diff
@@ -304,6 +309,30 @@ public ByteOrder byteOrderFor(IndexComponent indexComponent, IndexContext contex
         return ByteOrder.BIG_ENDIAN;
     }
 
+    @Override
+    public ByteComparable encodeForInMemoryTrie(ByteBuffer input, AbstractType<?> type)
```

Comment:
Why do we need different encodings for in memory tries depending on the version? Especially that `OnDiskFormat` still controls the conversion from in-memory to on disk encoding. This method feels to me like it doesn't belong to this place (it controls the on disk format not in memory format).

Reply:
I responded in the top discussion on this PR. Generally speaking, I think we have a tight coupling for the current on disk format and the in memory representation. Notably, for kd trees, we store the values in the trie in the same way we do on disk. Because that is currently the case, it seems reasonable to couple them together in the interface.

Reply:
Is it difficult to change this to always store the data in the trie in the same way as on disk?

When this was originally written the in-memory trie could not store prefixes (which necessitated the escaping), but this is not the case now.

Reply:
No, I don't think that would be too difficult. I will look into making that change.